runtime/v2/logging/logging_windows.go

```diff
@@ -0,0 +1,81 @@
+// +build windows
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package logging
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/Microsoft/go-winio"
+)
+
+// Run the logging driver
+func Run(fn LoggerFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var sout, serr net.Conn
+
+	if soutPipe := os.Getenv("CONTAINER_STDOUT"); soutPipe != "" {
+		locOut, err := winio.DialPipeContext(ctx, soutPipe)
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		sout = locOut
+	}
+
+	if serrPipe := os.Getenv("CONTAINER_STDERR"); serrPipe != "" {
+		locSerr, err := winio.DialPipeContext(ctx, serrPipe)
+
```

Comment:
Generally in Go you don't leave a newline between a statement and the error checking for that statement.